src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -157,7 +157,12 @@ class PersonalService(
       val maybeCurrentNonce = getCurrentAccount(request.from).map(_.nonce.toBigInt)
       val maybeNextTxNonce = maybeLatestPendingTxNonce.map(_ + 1) orElse maybeCurrentNonce
       val tx = request.toTransaction(maybeNextTxNonce.getOrElse(blockchainConfig.accountStartNonce))
-      val stx = wallet.signTx(tx)
+
+      val stx = if(blockchain.getBlockByNumber(blockchainConfig.eip155BlockNumber).isDefined){
+        wallet.signTx(tx, Some(blockchainConfig.chainId))
+      }else{
+        wallet.signTx(tx, None)
+      }
```

Comment:
Is this right? However unlikely this may be, say we're at a fork have the `X` block (where the EIP155 changeover happens), but a chain with a top block `X - 2` is better, so this TX will go into block `X - 1`...